pkg/server/test/list_objects.go

```diff
@@ -51,6 +52,63 @@ func newListObjectsRequest(store, objectType, relation, user, modelID string, co
 func ListObjectsTest(t *testing.T, ds storage.OpenFGADatastore) {
 	ctx := context.Background()
 
+	t.Run("list_objects_with_concurrent_checks_and_contextual_tuples", func(t *testing.T) {
```

Comment:
Added this here as a means to ensure we don't fail this issue again in the future. I'm working on a separate PR to add more functional tests for ListObjects like we've done with Check 👍 

In the future this test will move into that other test suite once it is done.